### PR TITLE
Conditional compilation for IR v7 support

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under OpenCV (BSD) License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/dnn.cpp [hunk 1]
```diff
@@ -3543,6 +3543,9 @@ Net Net::Impl::createNetworkFromModelOptimizer(InferenceEngine::CNNNetwork& ieNe
                 }
                 else
                 {
+#if INF_ENGINE_VER_MAJOR_GT(INF_ENGINE_RELEASE_2020_4)
+                    CV_Error(Error::StsNotImplemented, "This OpenCV version is built without Inference Engine IR v7 support");
```
Comment (dkurt) [inline modules/dnn/src/dnn.cpp:3547]: The message seems like there is an option to build with IRv7 support. I'd like to recommend to rephrase that support of old IRs has been removed.

Comment (ilya-lavrenov) [reply modules/dnn/src/dnn.cpp:3547]: It's just a copy-paste from the message for NN Builder :)

Comment (ilya-lavrenov) [reply modules/dnn/src/dnn.cpp:3547]: > The message seems like there is an option to build with IRv7 support

Just to pass older version of Inference Engine :)

Comment (ilya-lavrenov) [reply modules/dnn/src/dnn.cpp]: fixed

**Pull request merged**